src/components/Form/CrossPodWriteForm.jsx

```diff
@@ -20,69 +19,50 @@ import FormSection from './FormSection';
  *
  * @memberof Forms
  * @name CrossPodWriteForm
+ * @returns {React.ReactElement} The form
  */
-
 const CrossPodWriteForm = () => {
   const { session } = useSession();
   const { state, dispatch } = useStatusNotification();
-  const { selectedUser } = useContext(SelectUserContext);
-  const [username, setUsername] = useState();
   const [docType, setDocType] = useState('');
+  const [expiryDate, setExpiryDate] = useState('');
+  const [description, setDescription] = useState('');
+  const [file, setFile] = useState('');
+  const { podUrl, person } = useContext(SelectUserContext);
 
   const handleDocType = (event) => {
     setDocType(event.target.value);
   };
 
-  // Initalized state for file upload
-  const handleFileChange = (event) => {
-    if (event.target.files.length === 1) {
-      dispatch({ type: 'SET_FILE', payload: event.target.files[0] });
-    } else {
-      dispatch({ type: 'CLEAR_FILE' });
-    }
-  };
-
-  // Custom useField hook for handling form inputs
-  const { clearValue: clearDescription, _type, ...description } = useField('textarea');
-
-  const clearInputFields = (event) => {
-    event.target.reset();
-    clearDescription();
+  const clearInputFields = () => {
+    setDocType('');
+    setExpiryDate('');
+    setDescription('');
+    setFile(null);
     dispatch({ type: 'CLEAR_FILE' });
+    dispatch({ type: 'CLEAR_VERIFY_FILE' });
     dispatch({ type: 'CLEAR_PROCESSING' });
   };
 
-  const handleFormSubmit = makeHandleFormSubmission(
-    INTERACTION_TYPES.CROSS,
-    state,
-    dispatch,
-    session,
-    clearInputFields
-  );
-
   // Event handler for form/document submission to Pod
-  const handleCrossPodUpload = async (event) => {
-    event.preventDefault();
-    dispatch({ type: 'SET_PROCESSING' });
-    const podUsername = event.target.crossPodUpload.value || selectedUser.username;
-
-    if (!podUsername) {
-      runNotification('Search failed. Reason: Username not provided.', 5, state, dispatch);
-      setTimeout(() => {
-        dispatch({ type: 'CLEAR_PROCESSING' });
-      }, 3000);
-      return;
-    }
+  const handleFormSubmission = async (e) => {
+    e.preventDefault();
+    const fileDesc = {
+      name: file.name.split('.')[0],
+      type: docType,
+      date: expiryDate,
+      description
+    };
+    runNotification(`Uploading "${file.name}" to Solid...`, 3, state, dispatch);
 
-    if (!docType) {
-      runNotification('Search failed. Reason: No document type selected.', 5, state, dispatch);
-      setTimeout(() => {
-        dispatch({ type: 'CLEAR_PROCESSING' });
-      }, 3000);
-      return;
+    try {
+      await createDocument(file, fileDesc, session, `${podUrl}PASS/`);
+      runNotification(`File "${file.name}" updated on Solid.`, 5, state, dispatch);
+    } catch (error) {
+      runNotification(`File failed to upload. Reason: ${error.message}`, 5, state, dispatch);
```

Comment:
This this should trigger a second function if it's updating a file and it's not an upload error.

Reply:
This disabled file update. I don't think we actually have a use or UI for that in the system. It definitely shouldn't be in the same flow as upload.

Reply:
If you want to upload a new document, just upload a new document. If you want to replace a document, delete the old document, then upload a new one. If you want to update a doc's description, well we don't have a UI for that anyway right now.

Reply:
We had, that was part of updateDocument. If you were to change the description, it'll update the description in the document.ttl.

Reply:
Is that how the update flow is supposed to work? That's not how I found it working in testing. Here's what I tried doing in the current version (attempted action -> result): 

1. Upload a new file by selecting a type (passport) and a file (passport.pdf) to upload -> File and document.ttl are added to a pod
2. Try to update the file description by selecting the type passport and entering a new description (intending to target passport.pdf) -> cannot upload. upload button is disabled because there is no file added
3. Try to update a file description by selecting the type passport, entering a new description, and uploading passport.pdf again -> Prompted that the file already exists do I want to update? - > I click yes -> Description in document.ttl is updated, but it is unclear what uploading passport.pdf did. The system either deleted the original passport.pdf and replaced it with a new upload, or it ignored the new passport.pdf I added, and kept the original file the same.
4. Try to upload a new passport document `myNewPassport.pdf` with a description-> Get prompted that myNewPassport doesn't exist (I know?) do I want to upload? -> Click yes -> `myNewPassport.pdf` gets added to the pod. `document.ttl` gets updated with a new description, however the url field still points to `passport.pdf`, not `myNewPassport.pdf`. No new document.ttl is created for `myNewPassport.pdf`. The result is a document.ttl pointing to `passport.pdf` with a description intended for `myNewPassport.pdf`.

I'm not really sure what the intended flow is here. 

And besides, updating these documents (which are all actually collections of multiple documents and files) isn't a trivial problem. We can't update the files themselves (only delete and replace them), we can only update the descriptions we generate and store in document.ttl. However, a direct update of document.ttl not currently possible in the system (or if it is, it's too unintuitive to be useful). Also any updates to the documents invalidate signatures (at least how they're currently implemented). That should be taken into account when updating anything.

Reply:
How it's implemented now, especially after the initial upload, is, unfortunately, poorly-written on my part. However, I'll do my best to detailed what was intended before being moved to `/PASS/Documents/`. (see below)

---

If a user wishes to upload `document.pdf` to "Bank Statement" and the container for "Bank Statement" is empty, the user can simply:

1. Select `Bank Statement` as document type
2. Fill out the form
3. Choose the file
4. Click Upload document

Under the hood, the file will be:

5. Generate a new container at `<podUrl>Bank_Statement/`
5. Placed and save the file inside the container at `<podUrl>Bank_Statement/document.pdf`
6. Assigned the file will be assigned the URL `<podUrl>Bank_Statement/document.pdf` 
7. Generate a `document.ttl` file will be generated in that container.
8. Assign the container and `document.pdf` with default ACL settings for user.

If a user wishes to upload another document to the same directory, say "newDoc.pdf", to "Bank Statement", the user can follow the same 1-4 as before.

This time because the container is not empty, a function is called to alert the person that the container is not empty and the user could decide to either upload the new document or cancel the operation.

If the user cancels the operations. No actions will be taken. If the user wish to continue, then under the hood:

9. The file will be uploaded to `<podUrl>Bank_Statement/`
10. The file will be assigned the URL `<podUrl>Bank_Statement/newDoc.pdf` 
11. `document.ttl` will be appended by a new Thing instead of being generated using what's filled out in the form in 1-4 (this step is not implemented correctly unfortunate since it's uses .set for updating document description instead the .add functions)
12. Assign ACL for `document.pdf` with default ACL settings for user.

At this point there'll be three files in "Bank Statement" with `document.ttl` pointing to both `<podUrl>Bank_Statement/newDoc.pdf` and `<podUrl>Bank_Statement/document.pdf`

If a user wishes to update a file with the same name, say "document.pdf", the user would once again perform 1-4, but this time a function will be called alerting the user that "document.pdf" already exist in "Bank Statement" do they wish to continue. This part is the same as before with a new file, if user cancels the operation, nothing happens, if they wish to continue, then under the hood:

13. The file will be overwritten in `<podUrl>Bank_Statement/document.pdf`
14. `document.ttl` will be updated instead of generated using what's filled out in the form in 1-4 (this part for updating with .set is correct)

So the final result would have "Bank Statement" with `document.ttl` pointing to both `<podUrl>Bank_Statement/newDoc.pdf` and `<podUrl>Bank_Statement/document.pdf`, where `document.pdf` being updated and reflected in the `document.ttl`.

For the signatures, I could see it fitting in nicely before or after 5-8 and 9-12 for file uploads. If `signature.ttl` would need its own container, I could see us using sub containers with the file name and store that in there. But other than at I don't see any clashes for these two scenarios.

For 13-14 with file updates, I definitely see a need for re-validation as mentioned. By default, the original `signature.ttl` would be invalid because it's not associated with the updated file, so it'll have to be removed before update starts. From there, everything else should be the same as if a new file is being uploaded.

Reply:
Thinking about it more, we can actually get decent, stable functionality by just deleting and reuploading. Updated the PR to do that. You can now 'update' (delete and replace) files by uploading a file with the same name.